signer/src/context/messaging.rs

```diff
@@ -0,0 +1,70 @@
+//! This module contains types related to the application's internal
+//! messaging via the [`Context`].
+
+/// Signals that can be sent within the signer binary.
+#[derive(Debug, Clone)]
+pub enum SignerSignal {
+    /// Send a command to the application.
+    Command(SignerCommand),
+    /// Signal an event to the application.
+    Event(SignerEvent),
+}
+
+/// Commands that can be sent on the signalling channel.
+#[derive(Debug, Clone)]
+pub enum SignerCommand {
+    /// Signals to the application to publish a message to the P2P network.
+    P2PPublish(crate::network::Msg),
+}
+
+/// Events that can be received on the signalling channel.
+#[derive(Debug, Clone)]
+pub enum SignerEvent {
+    /// Signals that a P2P event has occurred.
+    P2P(P2PEvent),
+    /// Signals that the block observer database has been updated.
+    BlockObserverDbUpdated,
```

Comment:
Gah, these snuck in here. I can remove them if you want; they'll be added in a coming PR otherwise.